src/utils/hooks/use-token-info.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { createContext, useContext, useEffect, ReactNode } from 'react';
+import { useColonyNativeTokenQuery, useTokenInfoLazyQuery } from '~data/index';
+import { Address } from '~types/index';
+
+export type TokenInfo = {
+  colonyAddress: string;
+};
+
+const TokenInfoContext = createContext<Partial<TokenInfo>>({});
+
+export type Props = {
+  children: ReactNode;
+  colonyAddress: Address;
+};
+
+export const TokenInfoProvider = ({ children, colonyAddress }: Props) => (
+  <TokenInfoContext.Provider
+    value={{
+      colonyAddress,
+    }}
+  >
+    {children}
+  </TokenInfoContext.Provider>
+);
+
+export const useTokenInfo = () => {
```

Comment:
I'm good with this provider, but the file name is very **bad**

Please rename it to `userTokenInfo.ts`

Reply:
Also, make sure to export if from `index` too, so that we can import it like: 
```js
import { userTokenInfo } from '~utils/hooks';
```